usaspending_api/awards/v2/filters/award.py

```diff
@@ -1,204 +1,204 @@
+        # recipient_location_scope (broken till data reload)
         elif key == "recipient_scope":
-            if value is not None:
-                if value == "domestic":
-                    queryset = queryset.Award.filter(recipient__location__country_name="UNITED STATES")
-                elif value["type"] == "foreign":
-                    queryset = queryset.Transaction.exclude(recipient__location__country_name="UNITED STATES")
-                else:
-                    raise InvalidParameterException('Invalid filter: recipient_location type is invalid.')
+            if value == "domestic":
+                queryset = queryset.filter(recipient__location__country_name="UNITED STATES")
+            elif value == "foreign":
+                queryset = queryset.exclude(recipient__location__country_name="UNITED STATES")
+            else:
+                raise InvalidParameterException('Invalid filter: recipient_scope type is invalid.')
 
-        # recipient_location - DONE
+        # recipient_location
         elif key == "recipient_locations":
-            if value is not None:
-                or_queryset = Award.objects.none()
-                for v in value:
-                    or_queryset |= Award.filter(recipient__location__location_id=v)
-                queryset = queryset & or_queryset
-            else:
-                raise InvalidParameterException('Invalid filter: recipient_location object is invalid.')
+            or_queryset = None
+            for v in value:
+                if or_queryset:
+                    or_queryset |= or_queryset.filter(recipient__location__location_id=v)
+                else:
+                    or_queryset = Award.objects.filter(recipient__location__location_id=v)
+            if or_queryset is not None:
+                queryset &= or_queryset
 
-        # recipient_type_names - DONE
+        # recipient_type_names
         elif key == "recipient_type_names":
-            or_queryset = Award.objects.none()
+            or_queryset = None
             for v in value:
-                or_queryset |= Award.filter(recipient__business_types_description=v)
-            queryset &= or_queryset
+                if or_queryset:
+                    or_queryset |= or_queryset.filter(recipient__business_types_description=v)
+                else:
+                    or_queryset = Award.objects.filter(recipient__business_types_description=v)
+            if or_queryset is not None:
+                queryset &= or_queryset
 
-        # place_of_performance_scope (broken till data reload)- DONE
+        # place_of_performance_scope (broken till data reload
         elif key == "place_of_performance_scope":
             if value == "domestic":
-                queryset = queryset.Award.filter(place_of_performance__country_name="UNITED STATES")
+                queryset = queryset.filter(place_of_performance__country_name="UNITED STATES")
```

Comment:
This may need to have an `iexact`, if the format for the country name is anything besides `UNITED STATES` exactly, we may be overlooking some elements. For example, what would happen if the country name is lower cased? Is this something that is auto-populated by us during loads so we can control what it's set to?

Reply:
Awesome